### PR TITLE
Fixes game replays.

### PR DESCRIPTION
**Pull request by OmniBlade**
Makes recording and playback more or less work correctly when using the -XX and -XY command lines respectively.

### CODE REVIEW

File: redalert/sidebar.cpp [hunk 1]
```diff
@@ -975,7 +975,7 @@ bool SidebarClass::Activate(int control)
 
     bool old = IsSidebarActive;
 
-    if (Session.Play)
+    if (Session.Attract)
```
Comment (hifi) [inline redalert/sidebar.cpp]: Could fix the braces while you're at it.

File: tiberiandawn/sidebar.cpp [hunk 2]
```diff
@@ -1001,7 +1001,7 @@ bool SidebarClass::Activate(int control)
     int sidewidth = SeenBuff.Get_Width() - sidex;
     int sideheight = SeenBuff.Get_Height() - sidey;
 
-    if (PlaybackGame)
+    if (AllowAttract)
```
Comment (hifi) [inline tiberiandawn/sidebar.cpp]: Ditto for braces.

File: tiberiandawn/queue.cpp [hunk 3]
```diff
@@ -3150,6 +3150,7 @@ static void Queue_Playback(void)
     Compute_Game_CRC();
     CRC[Frame & 0x001f] = GameCRC;
 
+#if 0
```
Comment (hifi) [inline tiberiandawn/queue.cpp]: A comment why would be nice.

File: tiberiandawn/queue.cpp [hunk 4]
```diff
@@ -3287,7 +3288,7 @@ static void Compute_Game_CRC(void)
     //------------------------------------------------------------------------
     //	A random #
     //------------------------------------------------------------------------
-    Add_CRC(&GameCRC, rand());
+    // Add_CRC(&GameCRC, rand());
```
Comment (hifi) [inline tiberiandawn/queue.cpp:3293]: Comment here as well.

**Review by hifi — COMMENTED** [review]
Nice and simple changes. Does this also fix the desync issues with harvesters when playing a network game?

**Comment by OmniBlade** [comment]
It does fix the possible desync though in practice it won't when played on the same platform because rand will be the same and thus act deterministic. There is still an issue that rand is used in TD further up the chain of random calls, but I'll look to fix that with a RandomClass later.

**Review by hifi — COMMENTED** [review]
Now I've properly tested this and found a few issues. With RA credits don't tick up and the build queue doesn't react to building in the played back game. TD seems to work perfectly. You think you can fix that?

We could also expose this without CHEAT_KEYS as the only ifdeffed out bit is the command line parsing and it will definitely help testing it?

**Comment by OmniBlade** [comment]
@hifi I can't see any difference in behaviour for side bar building or credit updates between the games when I've retested replays. There are some graphical glitches when the terrain doesn't update properly when the tactical window moves that needs more work, but both games look equivalent to me?

I've updated the command line gating, command line parsing in general needs improving to move away from the special obfuscated options, but that is another piece of work.

**Comment by hifi** [comment]
RA still has some score and build queue issues that I can reliably reproduce by recording some building and TD has issues with the recording argument now and skips straight into game and bugs out.

**Comment by OmniBlade** [comment]
I can't reproduce these issues at all, -XX records and -XY plays back the recording correctly for both games for me (windows builds). Can you be more specific with what issues you are seeing with playback in RA?

**Comment by hifi** [comment]
Okay as it seems it's so random and I'm pretty sure I have `make clean` every time I say let's merge this and look at the issues separately if someone else can repro them later.

**Review by hifi — APPROVED** [review]

**Pull request merged**